src/api/cpp/backend/backend_engine.h

```diff
@@ -53,39 +54,35 @@ class nixlBackendEngine {
         }
 
     public:
-        nixlBackendEngine (const nixlBackendInitParams* init_params)
-            : localAgent(init_params->localAgent) {
-
-            this->backendType  = init_params->type;
-            this->initErr      = false;
-            this->customParams = new nixl_b_params_t(*(init_params->customParams));
+        explicit nixlBackendEngine (const nixlBackendInitParams* init_params)
+            : backendType( init_params->type ),
+              customParams( new nixl_b_params_t( *(init_params->customParams) ) ),  // TODO: Use std::make_unique (C++14).
```

Comment:
use make_unique 
extra () around init_params

Reply:
There are many places in NIXL where objects are manually allocated on heap and passed around as pointers instead of using them by-value or by-reference. We have been removing such cases and I suggest to also make customParams here a by-value class member instead of a (smart) pointer.